pysixdesk/lib/dbtable.py

```diff
@@ -0,0 +1,294 @@
+from collections import OrderedDict
+
+from . import dbtypedict
+
+class Table(object):
+    '''This class aims to initialize and customize the database tables'''
+
+    def __init__(self, tables, table_keys, db_type):
+        self.tables = tables
+        self.table_keys = table_keys
+        self.db_type = db_type
+        if db_type.lower() == 'sql':
+            self.type_dict = dbtypedict.SQLiteDict()
+        elif db_type.lower() == 'mysql':
+            self.type_dict = dbtypedict.MySQLDict()
+        self.tables['templates'] = OrderedDict()
+        self.tables['env'] = OrderedDict()
+        self.tables['boinc_vars'] = OrderedDict()
+        self.init_preprocess_tables()
+        self.init_sixtrack_tables()
+
+    @staticmethod
+    def result_table(filelist):
+        '''Retrun the map between the result files and database tables
+        fileName --> tableName'''
+        relation = {}
+        relation['fort.10'] = 'six_results'
+        relation['aperture_losses.dat'] = 'aperture_losses'
+        relation['oneturnresult'] = 'oneturn_sixtrack_results'
+        relation['Coll_Scatter.dat'] = 'collimation_losses'
+        relation['final_state.dat'] = 'final_state'
+        relation['initial_state.dat'] = 'init_state'
+
+        filemap = {}
+        for fil in filelist:
+            if fil in relation.keys():
+                filemap[fil] = relation[fil]
+            else:
+                filemap[fil] = None
+        return filemap
+
+    def customize_tables(self, table_names, info, dtype=None):
+        '''Customize the tables'''
+        if isinstance(info, dict):
+            for key, val in info.items():
+                self.tables[table_names][key] = self.type_dict[val]
+        elif isinstance(info, list):
+            if dtype is None:
+                raise Exception("Data type information needed!")
```

Comment:
Raising a `ValueError` is more accurate, than the unnecessarily broad `Exception`.

Reply:
Make sense!